lib/App/DuckPAN/Restart.pm

```diff
@@ -0,0 +1,104 @@
+package App::DuckPAN::Restart;
+# ABSTRACT: Automatic restarting of application on file change
+
+use File::Find::Rule;
+use Filesys::Notify::Simple;
+
+use strict;
+
+use Moo::Role;
+
+requires '_run_app';
+
+sub run_restarter {
+    my ($self, $args) = @_;
+
+    # will keep (re)starting the server until user ctrl-c
+    while(1){
+        defined(my $app = fork) or die 'Failed to fork application';
+        unless($app){ # app kid
+            $self->_run_app($args);
+            exit 0;
+        }
+
+        # Slightly different format here since we need to take care of
+        # the newly spawned app on failure.
+        my $fmon = fork;
+        unless($fmon){ # file monitor kid
+            unless(defined $fmon){
+                kill SIGTERM => $app;
+                die 'Failed to fork file monitor';
+            }
+            $self->_monitor_directories;
+            exit 0;
+        }
+
+        # wait for one them to exit. -1 waits for all children
+        my $pid = waitpid -1, 0;
```

Comment:
I think there might be a slight race condition here when *outside of an IA directory* if the pid for the fmon is reaped first.  Let me have a look at it.  I test this very condition this morning and it was fine but running it this afternoon for something unrelated triggered it.